src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesNoHttpTool.java

```diff
@@ -0,0 +1,159 @@
+package com.sap.sgs.phosphor.fosstars.data.github;
+
+import static com.sap.sgs.phosphor.fosstars.model.feature.oss.OssFeatures.USES_NOHTTP;
+
+import com.sap.sgs.phosphor.fosstars.model.Value;
+import com.sap.sgs.phosphor.fosstars.model.ValueSet;
+import com.sap.sgs.phosphor.fosstars.model.value.BooleanValue;
+import com.sap.sgs.phosphor.fosstars.tool.github.GitHubProject;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import org.apache.maven.model.Dependency;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.Plugin;
+import org.apache.maven.model.io.xpp3.MavenXpp3Reader;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+import org.kohsuke.github.GHContent;
+import org.kohsuke.github.GHFileNotFoundException;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GitHub;
+
+/**
+ * This data provider checks if an open-source project uses
+ * <a href="https://github.com/spring-io/nohttp">nohttp</a> tool to make sure that
+ * plain HTTP is not used.
+ * The provider fills out the
+ * {@link com.sap.sgs.phosphor.fosstars.model.feature.oss.OssFeatures#USES_NOHTTP} feature.
+ */
+public class UsesNoHttpTool extends AbstractGitHubDataProvider {
+
+  /**
+   * Initializes a data provider.
+   *
+   * @param github An interface to the GitHub API.
+   */
+  public UsesNoHttpTool(GitHub github) {
+    super(github);
+  }
+
+  @Override
+  protected UsesNoHttpTool doUpdate(GitHubProject project, ValueSet values) throws IOException {
+    logger.info("Figuring out if the project uses NoHTTP ...");
+
+    GHRepository repository = gitHubDataFetcher().repositoryFor(project, github);
+    boolean answer = checkMaven(repository) || checkGradle(repository);
+    Value<Boolean> value = new BooleanValue(USES_NOHTTP, answer);
+    values.update(value);
+
+    return this;
+  }
+
+  /**
+   * Checks if a project uses NoHTTP with Maven.
+   *
+   * @param repository The project's repository.
+   * @return True if the project uses the plugin, false otherwise.
+   */
+  private boolean checkMaven(GHRepository repository) throws IOException {
+    GHContent content;
+    try {
+      content = repository.getFileContent("pom.xml");
+    } catch (GHFileNotFoundException e) {
+      return false;
```

Comment:
An logger.warn() would be good here?

Reply:
I am not sure about that. If there is no `pom.xml` file, then it is okay because a project may use other build system.